R/ntfy.R

```diff
@@ -102,39 +140,19 @@ ntfy_send <- function(message  = "test",
     email    = email
   )
   payload <- Filter(Negate(is.null), payload)
-
-  if (!is.null(image)) {
-    send_image(trim_slash(server), topic, image, payload, config = auth, ...)
-  } else {
-    httr::POST(url = trim_slash(server), body = payload, encode = "json", config = auth, ...)
-  }
-}
-
-send_image <- function(server, topic, image, payload, ...) {
-  if (inherits(image, "ggplot")) {
-    requireNamespace("ggplot2", quietly = FALSE)
-    tmpf <- tempfile(pattern = "gg", fileext = ".png")
-    ggplot2::ggsave(tmpf, image)
-  } else if (is.character(image)) {
-    stopifnot(file.exists(image))
-    tmpf <- image
-  }
-  imagecon <- file(tmpf, "rb")
-  on.exit(close(imagecon))
-  imagebody <- readBin(con = imagecon,
-                       what = 'raw',
-                       n = file.info(tmpf)$size)
-  if (!is.null(payload$tags)) {
-    payload$tags <- toString(payload$tags)
-  }
-  httr::PUT(url = paste(server, topic, sep = "/"),
-            body = imagebody,
-            httr::add_headers(.headers = unlist(payload)), ...)
+  
+  resp <- httr2::request(server) |> 
+    httr2::req_url_path_append(topic) |> 
+    httr2::req_method("POST") |> 
+    httr2::req_user_agent("ntfy (https://github.com/jonocarroll/ntfy)") |> 
```

Comment:
Thank you for adding this!

Reply:
Thanks to Hadley for the guide to wrapping APIs :) https://httr2.r-lib.org/articles/wrapping-apis.html